tests/eas/hmp_parity.py

```diff
@@ -400,108 +423,92 @@ def test_first_cpu_migrators(self):
-            self.m_assert.getResidency(
-                "cluster",
-                self.env.target.bl.bigs,
-                percent=True,
-                window=big_residency_window
-            ), self.log_fh)
-
-        self.assertTrue(
-            self.m_assert.assertResidency(
-                "cluster",
-                self.env.target.bl.bigs,
-                EXPECTED_RESIDENCY_PCT,
-                operator.ge,
-                percent=True,
-                window=big_residency_window,
-                rank=self.num_tasks),
-            msg="Not all 'migrator' tasks are running on big cores for at least {}% of their execution time"\
-                    .format(EXPECTED_RESIDENCY_PCT))
-
-
-    def test_migrators_switch(self):
-        "Offload Migration and Idle Pull: Test LITTLE -> BIG Idle Pull Switch (Migrators)"
-        switch_window = (self.offset + 4.5, self.offset + 5.5)
-
-        logging.info(
-            "Offload Migration and Idle Pull: Test LITTLE -> BIG Idle Pull Switch (Migrators)")
-        log_result(
-            self.m_assert.assertSwitch(
-                "cluster",
-                self.env.target.bl.littles,
-                self.env.target.bl.bigs,
-                window=switch_window), self.log_fh)
-
-        self.assertTrue(
-            self.m_assert.assertSwitch(
-                "cluster",
-                self.env.target.bl.littles,
-                self.env.target.bl.bigs,
-                window=switch_window,
-                rank=self.num_tasks),
-            msg="Not all 'migrator' tasks are pulled by idle big cores when expected")
-
-
-    def test_big_res_early_starters(self):
-        """Offload Migration and Idle Pull: Test Big Residency (EarlyStarters)"""
-        logging.info(
-            "Offload Migration and Idle Pull: Test Big Residency (EarlyStarters)")
-
-        big_residency_window = (self.offset, self.offset + 5)
-
-        log_result(self.e_assert.getResidency(
-            "cluster",
-            self.env.target.bl.bigs,
-            percent=True,
-            window=big_residency_window), self.log_fh)
-
-        self.assertTrue(
-            self.e_assert.assertResidency(
-                "cluster",
-                self.env.target.bl.bigs,
-                EXPECTED_RESIDENCY_PCT,
-                operator.ge,
-                percent=True,
-                window=big_residency_window,
-                rank=self.num_tasks),
-            msg="Not all 'early starter' tasks are running on big cores for at least {}% of their execution time"\
-                    .format(EXPECTED_RESIDENCY_PCT))
+    def test_big_cpus_fully_loaded(self):
+        """Offload Migration and Idle Pull: Big cpus are fully loaded as long as there are tasks left to run in the system"""
+        num_big_cpus = len(self.env.target.bl.bigs)
+
+        end_times = sorted(self.end_times.values())
+
+        # Window of time until the first migrator finishes
+        window = (0, end_times[-num_big_cpus])
```

Comment:
No, there is no assumption in the ordering of big and LITTLE cpu in `end_times`.  The `end_times` array holds the times at which the tasks finished.  The window is the window from start until the first migrator finishes.  We then make sure that the bigs were fully utilized during this time.  We are not checking which task ran on a big cpu, just that while there are more tasks than bigs still running, the bigs are fully utilized.